pennylane/ops/qubit/state_preparation.py

```diff
@@ -227,7 +227,7 @@ def state_vector(self, wire_order=None):
             ket_shape = [self.batch_size] + ket_shape
 
         ket = np.zeros(ket_shape, dtype=np.complex128)
-        ket[indices] = op_vector
+        ket[indices] = math.unwrap(op_vector)
```

Comment:
Wouldn't this break differentiation?

Reply:
not anymore! totally re-vamped the function.